contracts/DomainRegistry.sol

```diff
@@ -48,65 +37,58 @@ contract DomainRegistry {
         _;
     }
 
-modifier parentDomainExists(string memory domainName) {
-    if (isTopLevelDomain(domainName)) {
+    modifier domainExistence(string memory domainName, bool shouldExist) {
+        bool exists = domains[domainName.stripProtocol()].owner != address(0);
+        if (shouldExist) {
+            require(exists, "Domain doesn't exist");
+        } else {
+            require(!exists, "Domain exists");
+        }
         _;
-        return;
-    }
-    
-    string memory parentDomain = domainName.extractParentDomain();
-    if (bytes(parentDomain).length > 1) {  
-        require(domains[parentDomain].isRegistered, "Parent domain doesn't exist");
     }
-    _;
-}
 
-function isTopLevelDomain(string memory domainName) internal pure returns (bool) {
-    bytes memory domainBytes = bytes(domainName);
-    uint dotCount = 0;
-    for (uint i = 0; i < domainBytes.length; i++) {
-        if (domainBytes[i] == bytes1('.')) {
-            dotCount++;
-        }
+    function setDomainPrice(uint256 _price) external onlyOwner {
+        domainPrice = _price;
     }
-    return dotCount == 0;  
-}
+
 
 function registerDomain(string memory domainName) 
     public 
-    hasRequiredDeposit() 
-    domainExistence(domainName, false) 
-    isValidDomain(domainName)
-    parentDomainExists(domainName)
     payable 
+    isValidDomain(domainName)
+    domainExistence(domainName, false)
 {
+    require(msg.value == domainPrice, "Incorrect amount sent");
+
     domainName = domainName.stripProtocol();
```

Comment:
You stripProtocol() three times: once in each of the modifiers and once here. Consider better code design.
-1 point